### PR TITLE
Update example to use Node's native HTTP module

### PR DESCRIPTION
**Pull request by vaurdan**
The goal of this PR is to provide an example HTTP server without using external modules, such as Express.

The server was implemented using the requirements set by the previous Express example: 
 * A very simple response to `GET /`
 * Health check response on `GET /cache-healthcheck`
 * 404 responses otherwise.

As an (intended) side-effect, this PR removes the `express` package. 

## How to test

1. Checkout this PR.
2. Run:
```
npm install
npm run build
npm start
```
3. Check http://localhost:3000. You should see `Howdy!`.
4. Check http://localhost:3000/cache-healthcheck. You should see an empty 200 response.

### CODE REVIEW

**Review by chriszarate — CHANGES_REQUESTED** [review]
Nice! I let a few comments.

**Comment by vaurdan** [comment]
@chriszarate thanks for the feedback! I just submitted some changes that should address what you flagged:

I have removed the `GET` validation and instead, it's now validating alongside the request path. In this example, both endpoints will be "listening" for `GET` and `HEAD` verbs only. This is a very basic example, let me know if you think it would be best to create some helper functions to map callbacks to a given path and HTTP method (eg, `get( path, callback )`,`post( path, callback )`, etc). 

Regarding the second suggestion, I have changed that code to use the `URL` object instead of the manually parsing the request URL, as suggested. However, it is still trimming the trailing slash, if it exists. This helps, for example, to match both `localhost:3000/example` and `localhost:3000/example/` to the same endpoint without the need for an additional condition.

File: src/index.js [hunk 1]
```diff
@@ -1,18 +1,36 @@
-var express = require( 'express' );
-var app = express();
+const http = require( 'http' );
+
 const PORT = process.env.PORT || 3000;
+const BASEURL = process.env.BASEURL || 'http://localhost/';
```
Comment (chriszarate) [inline src/index.js:4]: Great idea 👍

File: src/index.js [hunk 2]
```diff
@@ -1,18 +1,36 @@
...
-// https://docs.wpvip.com/technical-references/vip-platform/node-js/
-app.get( '/cache-healthcheck?', function( req, res ) {
-	res.sendStatus( 200 );
-} );
+const app = http.createServer( ( req, res ) => {
+	let { pathname: requestPath } = new URL( req.url, BASEURL );
+
+	// Remove trailing slash from the request path, if exists.
+	requestPath = ( requestPath.length > 1 && requestPath.endsWith( '/' ) ) ? requestPath.slice( 0, -1 ) : requestPath;
+
```
Comment (chriszarate) [inline src/index.js:11]: > However, it is still trimming the trailing slash, if it exists. This helps, for example, to match both localhost:3000/example and localhost:3000/example/ to the same endpoint without the need for an additional condition.

I appreciate this approach and I think it makes sense in the context of a "real" application. Previously, the goal of this skeleton may have been to serve as a building block for such an application. However, we've seen that no one really uses this code. Instead they use a framework that implements routing in an opinionated way. Therefore, this is really just a generic example of how to implement a cache-healthcheck that keeps the site healthy until the customer commits their own code.

I'd remove this just to keep it simple. Note that we will never send `/cache-healthcheck/`, we always omit the trailing slash.

```suggestion
```

File: src/index.js [hunk 3]
```diff
@@ -1,18 +1,36 @@
...
+const BASEURL = process.env.BASEURL || 'http://localhost/';
 
-// Used by the monitoring system on VIP to verify the health of the app
-// Should return 200 when the app is healthy
-// https://docs.wpvip.com/technical-references/vip-platform/node-js/
-app.get( '/cache-healthcheck?', function( req, res ) {
-	res.sendStatus( 200 );
-} );
+const app = http.createServer( ( req, res ) => {
+	let { pathname: requestPath } = new URL( req.url, BASEURL );
```
Comment (chriszarate) [inline src/index.js:7]: Per change below

```suggestion
	const { pathname: requestPath } = new URL( req.url, BASEURL );
```

File: src/index.js [hunk 4]
```diff
@@ -1,18 +1,36 @@
...
+		return;
+	}
+
+	res.writeHead( 404 );
+	res.end( );
 } );
 
-app.listen( PORT, function() {
+
+app.on( 'listening', () => {
```
Comment (chriszarate) [inline src/index.js:32]: Neat, I didn't know you could "listen for listening" :)

File: src/index.js (hunk 2 above)
Comment (vaurdan) [reply src/index.js:11]: Thanks, that makes perfect sense!

**Review by chriszarate — APPROVED** [review]

**Pull request merged**